guard/src/commands/validate.rs

```diff
@@ -128,64 +148,106 @@ Note - When pointing the command to a directory, the directory may not contain a
-                .multiple(true)
                 .conflicts_with("payload"))
-            .arg(Arg::with_name(DATA.0).long(DATA.0).short(DATA.1).takes_value(true)
+            .arg(Arg::new(DATA.0)
+                .long(DATA.0)
+                .short(DATA.1)
+                .num_args(0..)
+                .action(ArgAction::Append)
                 .help("Provide a data file or directory of data files in JSON or YAML. Supports passing multiple values by using this option repeatedly.\
                           \nExample:\n --data template1.yaml --data ./data-dir1 --data template2.yaml\
                           \nFor directory arguments such as `data-dir1` above, scanning is only supported for files with following extensions: .yaml, .yml, .json, .jsn, .template")
-                .multiple(true)
                 .conflicts_with("payload"))
-            .arg(Arg::with_name(INPUT_PARAMETERS.0).long(INPUT_PARAMETERS.0).short(INPUT_PARAMETERS.1).takes_value(true)
+            .arg(Arg::new(INPUT_PARAMETERS.0)
+                .long(INPUT_PARAMETERS.0)
+                .short(INPUT_PARAMETERS.1)
+                .num_args(0..)
+                .action(ArgAction::Append)
                 .help("Provide a data file or directory of data files in JSON or YAML that specifies any additional parameters to use along with data files to be used as a combined context. \
                            All the parameter files passed as input get merged and this combined context is again merged with each file passed as an argument for `data`. Due to this, every file is \
                            expected to contain mutually exclusive properties, without any overlap. Supports passing multiple values by using this option repeatedly.\
                           \nExample:\n --input-parameters param1.yaml --input-parameters ./param-dir1 --input-parameters param2.yaml\
-                          \nFor directory arguments such as `param-dir1` above, scanning is only supported for files with following extensions: .yaml, .yml, .json, .jsn, .template")
-                .multiple(true))
-            .arg(Arg::with_name(TYPE.0).long(TYPE.0).short(TYPE.1).takes_value(true).possible_values(["CFNTemplate"])
-                .help("Specify the type of data file used for improved messaging"))
-            .arg(Arg::with_name(OUTPUT_FORMAT.0).long(OUTPUT_FORMAT.0).short(OUTPUT_FORMAT.1).takes_value(true)
-                .possible_values(["json", "yaml", "single-line-summary"])
+                          \nFor directory arguments such as `param-dir1` above, scanning is only supported for files with following extensions: .yaml, .yml, .json, .jsn, .template"))
+            .arg(Arg::new(TYPE.0)
+                .long(TYPE.0)
+                .short(TYPE.1)
+                .action(ArgAction::Set)
+                .required(false)
+                .help("Specify the type of data file used for improved messaging - ex: CFNTemplate"))
```

Comment:
We have moved `possible_values(["CFNTemplate"])` part to the string to enum processing snippet haven't we?

Shouldn't we have a similar `.value_parser(TYPE_BLAH_MEH)` here?

Reply:
I dont think thats the correct behaviour since if we had something like 

`TYPES_VALUES: [&str: 1] = ["CFNTemplate"]`

we would error out when any other value was passed, instead if any other value is passed we use the `generic` handler